packages/appium/lib/extension/manifest.js

```diff
@@ -175,26 +160,58 @@ export class Manifest {
 
   /**
    * Searches `APPIUM_HOME` for installed extensions and adds them to the manifest.
-   * @param {SyncWithInstalledExtensionsOpts} opts
    * @returns {Promise<boolean>} `true` if any extensions were added, `false` otherwise.
    */
-  async syncWithInstalledExtensions({depthLimit = DEFAULT_SEARCH_DEPTH} = {}) {
-    const walkOpts = _.defaults({depthLimit}, DEFAULT_FIND_EXTENSIONS_OPTS);
+  async syncWithInstalledExtensions() {
     // this could be parallelized, but we can't use fs.walk as an async iterator
     let didChange = false;
-    for await (const {stats, path: filepath} of fs.walk(this._appiumHome, walkOpts)) {
-      if (filepath !== this._appiumHome && stats.isDirectory()) {
-        try {
-          const pkg = await env.readPackageInDir(filepath);
-          if (pkg && isExtension(pkg)) {
-            // it's possible that this extension already exists in the manifest,
-            // so only update `didChange` if it's new.
-            const added = this.addExtensionFromPackage(pkg, path.join(filepath, 'package.json'));
-            didChange = didChange || added;
+
+    /**
+     * Listener for the `match` event of a `glob` instance
+     * @param {string} filepath - Path to a `package.json`
+     * @returns {Promise<void>}
+     */
+    const onMatch = async (filepath) => {
+      try {
+        const pkg = JSON.parse(await fs.readFile(filepath, 'utf8'));
+        if (isDriver(pkg) || isPlugin(pkg)) {
+          const changed = this.addExtensionFromPackage(pkg, filepath);
+          didChange = didChange || changed;
+        }
+      } catch {}
```

Comment:
when node started to support this syntax?

Reply:
I don't recall, but it does.